python/config_settings/transition.bzl

```diff
@@ -14,266 +14,15 @@
-    "_collect_cc_coverage": attr.label(
-        default = "@bazel_tools//tools/test:collect_cc_coverage",
-        executable = True,
-        cfg = "exec",
-    ),
-    # Magic attribute to make coverage work. There's no
-    # docs about this; see TestActionBuilder.java
-    "_lcov_merger": attr.label(
-        default = configuration_field(fragment = "coverage", name = "output_generator"),
-        executable = True,
-        cfg = "exec",
-    ),
-}
-
-_transition_py_binary = rule(
-    _transition_py_impl,
-    attrs = _COMMON_ATTRS | _PY_TEST_ATTRS,
-    cfg = _transition_python_version,
-    executable = True,
-    fragments = ["py"],
-)
-
-_transition_py_test = rule(
-    _transition_py_impl,
-    attrs = _COMMON_ATTRS | _PY_TEST_ATTRS,
-    cfg = _transition_python_version,
-    test = True,
-    fragments = ["py"],
-)
-
-def _py_rule(rule_impl, transition_rule, name, python_version, **kwargs):
-    pyargs = py_args(name, kwargs)
-    args = pyargs["args"]
-    data = pyargs["data"]
-    env = pyargs["env"]
-    srcs = pyargs["srcs"]
-    deps = pyargs["deps"]
-    main = pyargs["main"]
-
-    # Attributes common to all build rules.
-    # https://bazel.build/reference/be/common-definitions#common-attributes
-    compatible_with = kwargs.pop("compatible_with", None)
-    deprecation = kwargs.pop("deprecation", None)
-    exec_compatible_with = kwargs.pop("exec_compatible_with", None)
-    exec_properties = kwargs.pop("exec_properties", None)
-    features = kwargs.pop("features", None)
-    restricted_to = kwargs.pop("restricted_to", None)
-    tags = kwargs.pop("tags", None)
-    target_compatible_with = kwargs.pop("target_compatible_with", None)
-    testonly = kwargs.pop("testonly", None)
-    toolchains = kwargs.pop("toolchains", None)
-    visibility = kwargs.pop("visibility", None)
-
-    common_attrs = {
-        "compatible_with": compatible_with,
-        "deprecation": deprecation,
-        "exec_compatible_with": exec_compatible_with,
-        "exec_properties": exec_properties,
-        "features": features,
-        "restricted_to": restricted_to,
-        "target_compatible_with": target_compatible_with,
-        "testonly": testonly,
-        "toolchains": toolchains,
-    }
-
-    # Test-specific extra attributes.
-    if "env_inherit" in kwargs:
-        common_attrs["env_inherit"] = kwargs.pop("env_inherit")
-    if "size" in kwargs:
-        common_attrs["size"] = kwargs.pop("size")
-    if "timeout" in kwargs:
-        common_attrs["timeout"] = kwargs.pop("timeout")
-    if "flaky" in kwargs:
-        common_attrs["flaky"] = kwargs.pop("flaky")
-    if "shard_count" in kwargs:
-        common_attrs["shard_count"] = kwargs.pop("shard_count")
-    if "local" in kwargs:
-        common_attrs["local"] = kwargs.pop("local")
-
-    # Binary-specific extra attributes.
-    if "output_licenses" in kwargs:
-        common_attrs["output_licenses"] = kwargs.pop("output_licenses")
-
-    rule_impl(
```

Comment:
Does this mean that we will have fewer targets? The way I read the code is that the new version does not add this extra private target.

Reply:
Correct.